apps/backoffice-v2/src/lib/blocks/hooks/useEntitiesDocuments/useEntitiesDocuments.ts

```
@@ -0,0 +1,119 @@
+import { TWorkflowById } from '@/domains/workflows/fetchers';
+import { selectDirectors } from '@/pages/Entity/selectors/selectDirectors';
+import { TDocument } from '@ballerine/common';
+import { useCallback, useMemo } from 'react';
+
+export type TEntitiesDocumentsFilter = 'all' | 'directors' | 'ubos';
+
+const allOrExact = (filter: TEntitiesDocumentsFilter, exact: TEntitiesDocumentsFilter) =>
+  filter === 'all' || filter === exact;
+
+export const useEntitiesDocuments = (
+  workflow?: TWorkflowById,
+  filter: 'all' | 'directors' | 'ubos' = 'all',
+) => {
+  const directors = useMemo(
+    () => (allOrExact(filter, 'directors') ? selectDirectors(workflow) || [] : []),
+    [workflow, filter],
+  );
+  const ubos = useMemo(
+    () => (allOrExact(filter, 'ubos') ? workflow?.childWorkflows || [] : []),
+    [workflow, filter],
+  );
+
+  const entityIds = useMemo(
+    () =>
+      [
+        ...directors.map(director => director.ballerineEntityId),
+        ...ubos.map(ubo => ubo.entity.id),
+        filter === 'all' ? workflow?.context?.entity?.ballerineEntityId : undefined,
+      ]
+        .filter(Boolean)
+        .flat(),
+    [directors, ubos, workflow, filter],
+  );
+  const contextDocuments = useMemo(() => workflow?.context?.documents || [], [workflow]);
+
+  const { documents, documentsSchemas, isLoading } = useDocumentsAdapter({
+    entityIds,
+    documents: contextDocuments as TDocument[],
+  });
+
+  const getDocumentEntityType = useCallback(
+    (document: TDocument) => {
+      if (document.businessId === workflow?.context?.entity?.ballerineEntityId) {
+        return 'business';
+      }
+
+      if (directors.some(director => director.ballerineEntityId === document.endUserId)) {
+        return 'director';
+      }
+
+      if (ubos.some(ubo => ubo.entity.id === document.endUserId)) {
+        return 'ubo';
+      }
+
+      return document.type;
```

Is this intentional? Or is it supposed to be some specific value like identification_documents or should we throw an error?

---

Its intentional, just selector for string which is being rendered. No need to throw here.